addons/loyalty_delivery/tests/test_loyalty_delivery.py

```diff
@@ -18,6 +20,70 @@ def setUpClass(cls):
         cls.product_4 = cls.env['product.product'].create({'name': 'A product to deliver'})
         cls.product_uom_unit = cls.env.ref('uom.product_uom_unit')
 
+    def test_free_delivery_cost_with_ewallet(self):
+        """
+        Test that paying a sale order with an ewallet does not affect when free shipping is
+        applied if the free shipping depends on the amount of the order.
```

Comment:
I'm sure you can find a clearer docstring :eyes:.
"Test that" has no meaning, it's a test method.
"... when ... if ..." is quite heavy, prefer clear ordered steps like the next method docstring.